bgpd/bgp_main.c

```diff
@@ -469,6 +476,23 @@ main (int argc, char **argv)
 	  if (vty_port <= 0 || vty_port > 0xffff)
 	    vty_port = BGP_VTY_PORT;
 	  break;
+	case OPTION_VTYSOCK:
+	  vty_sock_name = strrchr (BGP_VTYSH_PATH, '/');
```

Comment:
OPTION_VTYSOCK functionality cut-n-pasted throughout each daemon.  Create a function to live in lib/sockopt.c?